aim2dat/strct/strct_manipulation.py

```diff
@@ -120,9 +120,52 @@ def substitute_elements(
         return _add_label_suffix(new_structure, "_subst-" + "-".join(str_el_pairs), change_label)
 
 
-def scale_unit_cell(structure: Structure, scaling_factor: float, change_label: bool) -> Structure:
-    """Scale unit cell of the structure."""
-    if scaling_factor != 1.0 and structure["cell"] is not None:
-        new_structure = structure.to_dict(cartesian=False)
-        new_structure["cell"] = np.array(structure["cell"]) * scaling_factor
-        return _add_label_suffix(new_structure, f"_scaled-{scaling_factor}", change_label)
+def scale_unit_cell(
+    structure,
+    scaling_factors: Union[float, List[float], np.ndarray] = None,
+    pressure: float = None,
+    bulk_modulus: float = None,
+    strain: Union[float, np.ndarray, list] = None,
+    change_label: bool = True,
+) -> Structure:
+    """Scale the unit cell of a structure."""
+    from aim2dat.strct import Structure
+
+    def get_strain_tensor(strain):
+        """Convert strain input into a 3x3 strain tensor."""
+        if isinstance(strain, (float, int)):
+            return np.eye(3) * (1 + strain)
+        elif isinstance(strain, list) and len(strain) == 3:
+            return np.diag([1 + s for s in strain])
+        elif isinstance(strain, np.ndarray) and strain.shape == (3, 3):
+            return np.eye(3) + strain
```

Comment:
I think it's a bit confusing to distinguish between `list`s and `np.array`s like this. If we allow both of them `strain: Union[float, np.ndarray, list]`, I'd suggest to change the `get_strain_tensor` logic. Anisotropic Strain can be provided as a `list` or 1D-`np.array` of length 3. Similarly, a `3x3` strain tensor can be provided as a nested `list` or 2D-`np.array`.

Reply:
Thanks a lot for your feedback! I modified the function to make it simpler and more concise.